lib/new_relic/telemetry/broadway.ex

```diff
@@ -0,0 +1,190 @@
+defmodule NewRelic.Telemetry.Broadway do
+  @moduledoc false
+  use GenServer
+
+  alias NewRelic.Transaction
+  alias NewRelic.DistributedTrace
+
+  require Logger
+
+  @processor_start [:broadway, :processor, :start]
+  @processor_stop [:broadway, :processor, :stop]
+
+  @message_stop [:broadway, :processor, :message, :stop]
+  @message_fail [:broadway, :processor, :message, :failure]
+
+  @consumer_start [:broadway, :consumer, :start]
+  @consumer_stop [:broadway, :consumer, :stop]
+
+  def start_link() do
+    enabled = true
+    GenServer.start_link(__MODULE__, [enabled: enabled], name: __MODULE__)
+  end
+
+  def init(enabled: false), do: :ignore
+
+  def init(enabled: true) do
+    config = %{
+      handler_id: :new_relic_broadway
+    }
+
+    :telemetry.attach_many(
+      config.handler_id,
+      [
+        @processor_start,
+        @processor_stop,
+        @message_stop,
+        @message_fail,
+        @consumer_start,
+        @consumer_stop
+      ],
+      &__MODULE__.handle_event/4,
+      config
+    )
+
+    Process.flag(:trap_exit, true)
+    {:ok, config}
+  end
+
+  def terminate(_reason, %{handler_id: handler_id}) do
+    :telemetry.detach(handler_id)
+  end
+
+  def handle_event(@processor_start, _measurements, metadata, _config) do
```

Comment:
How long does the "processor" live? If it doesn't have a short lifecycle (from seconds to a few mins), we can't use it as the Transaction, as it'd just build up memory and risk OOM kills

Reply:
The processor process can leave as long the app lives, but this transaction runs for each "batch". I'm running this fork on my app for testing, there are ~10 minute long transactions. If you wish, I can send you an email with a link for you to access this app.

Reply:
Change strategy, single Transaction per `handle_message/3`. This issue can appear on Consumer/`handle_batch/4` but I don't think there is much to do there - that's what it does, process a batch of messages.

I'll mark this as resolved, but please let me know if you have more thoughts around this strategy